cclib/parser/cfourparser.py

```diff
@@ -0,0 +1,619 @@
+        # get the name of the basis set used
+        if "BASIS                IBASIS" in line:
+            self.metadata["basis_set"] = line.split()[2]
+        # get calc_level
+        if "CALCLEVEL            ICLLVL" in line:
+            self.set_attribute("calc_level", line.strip().split()[2])
+        # get excited_states_method
+        if "EXCITE               IEXCIT" in line:
+            if not line.strip().split()[2] == "NONE":
+                self.metadata["excited_states_method"] = (
+                    line.strip().split()[2] + "-" + self.calc_level
+                )
+        # get whether the reference is unrestricted or not
+        if "REFERENCE            IREFNC" in line:
+            if line.split()[2][0] == "U":
+                self.metadata["unrestricted"] = True
+                self.set_attribute("homos", np.array([0, 0]))
+            else:
+                if line.split()[2] == "ROHF":
+                    self.set_attribute("homos", np.array([0, 0]))
+                else:
+                    self.set_attribute("homos", np.array([0]))
+                self.metadata["unrestricted"] = False
+        # get full point group
+        if "The full molecular point group is" in line:
+            self.metadata["symmetry_detected"] = line.split()[6].lower()
+        # get used point group
+        if "The computational point group is" in line:
+            self.metadata["symmetry_used"] = line.split()[5].lower()
+        # get success, cpu time, and wall time
+        if "@CHECKOUT-I, Total execution time (CPU/WALL):" in line:
+            self.metadata["success"] = True
+            self.metadata["cpu_time"] = [timedelta(seconds=float(line.strip().split()[5][:-1]))]
+            self.metadata["wall_time"] = [timedelta(seconds=float(line.strip().split()[6]))]
+        # get the net charge of the system
+        if "CHARGE               ICHRGE" in line:
+            self.set_attribute("charge", utils.float(line.split()[2]))
+        # get estate_prop state
+        if "ESTATE_PROP          IEXPRP" in line:
+            if line.strip().split()[2] == "OFF":
+                self.set_attribute("estate_prop_on", False)
+            else:
+                self.set_attribute("estate_prop_on", True)
+        # get the spin multiplicity of the system
+        if "MULTIPLICTY          IMULTP" in line:
+            self.set_attribute("mult", int(line.split()[2]))
+        # get coupled cluster energy
+        if "A miracle has come to pass. The CC iterations have converged." in line:
+            cc_lines = []
+            while "@CHECKOUT-I," not in line:
+                if not line.strip() == "":
+                    cc_lines.append(line)
+                line = next(inputfile)
+            if cc_lines[-1].split()[-1] == "a.u.":
+                self.ccenergies.append(utils.float(cc_lines[-1].split()[-2]))
+            else:
+                self.ccenergies.append(utils.float(cc_lines[-1].split()[-1]))
```

Comment:
Same for a branch like this:
```suggestion
            tokens = cc_lines[-1].split()
            if tokens[-1] == "a.u.":
                ccenergy_index = -2
            else:
                ccenergy_index = -1
            self.append_attribute("ccenergies", tokens[ccenergy_index])
```